django/db/models/functions/json.py

```diff
@@ -9,25 +11,40 @@ class JSONArray(Func):
     function = "JSON_ARRAY"
     output_field = JSONField()
 
+    class OnNull(enum.Enum):
+        NULL = "NULL ON NULL"
+        ABSENT = "ABSENT ON NULL"
+
+    def __init__(self, *expressions, on_null=OnNull.NULL):
```

Comment:
We've opted to use boolean flags instead for these kind of options (e.g. see `OrderBy(nulls_last)`, UniqueConstraint(nulls_distinct)` and others).

```suggestion
    def __init__(self, *expressions, absent_on_null=False):
```

Not sure about what the kwarg should be named though. It doesn't have to be a 1:1 map with the SQL feature name so maybe `exclude_nulls=False` could do?

Reply:
Happy to use a bool if you think that's best. I sort of start with SQL and work my way backwards.

Reply:
I've got a top-level comment, but I'm not sure the best way to proceed if we want to also include JsonObject. I think it probably would be best to try and include JsonObject, especially because I don't want to introduce TWO features or repurpose a feature that explicitly has the word `array` in it.